### PR TITLE
feat: always use new session when calling connect method for WC

### PR DESCRIPTION
**Pull request by MananTank**
## PR-Codex overview
This PR enhances wallet functionality by adding `accountsChanged` event, updating disconnect methods to return promises, and improving JSDoc consistency.

### Detailed summary
- Updated `DisconnectFn` to return a promise in `types.ts`
- Added `accountsChanged` event in `WalletEmitterEvents`
- Updated disconnect methods in various wallet files to return promises
- Renamed `useConnectedAccounts` to `useConnectedWallets`
- Updated JSDoc comments for clarity

> ✨ Ask PR-Codex anything about this PR by commenting with `/codex {your question}`

### CODE REVIEW

**Comment by MananTank** [comment]
TODO: figure out whether or not to remote the `if (provider.session)` condition or not after testing

File: .changeset/wise-tigers-drive.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+"thirdweb": patch
+---
+
+Always start a new session when calling `wallet.connect` for WalletConnect
```
Comment (jnsdls) [inline .changeset/wise-tigers-drive.md:5]: lets also call out the other changes please

File: packages/thirdweb/src/wallets/coinbase/coinbaseSDKWallet.ts [hunk 2]
```diff
@@ -235,6 +232,7 @@ function onConnect(
         address: getAddress(accounts[0]),
       };
       emitter.emit("accountChanged", newAccount);
+      emitter.emit("accountsChanged", accounts);
```
Comment (jnsdls) [inline packages/thirdweb/src/wallets/coinbase/coinbaseSDKWallet.ts:235]: just a comment, these are pretty close but do pretty different things, I wonder if we need the `accountChanged`

not a blocker but we should revisit this

File: packages/thirdweb/src/wallets/create-wallet.ts [hunk 3]
```diff
@@ -76,10 +76,15 @@ export function createWallet<const ID extends WalletId>(
         chain = newChain;
       });
 
-      const unsubscribeDisconnect = emitter.subscribe("disconnect", () => {
+      const reset = () => {
         account = undefined;
         chain = undefined;
-        // unsubscribe
+      };
```
Comment (jnsdls) [inline packages/thirdweb/src/wallets/create-wallet.ts:82]: nit: prefer `function reset() {}`

File: packages/thirdweb/src/wallets/wallet-connect/index.ts [hunk 4]
```diff
@@ -70,10 +70,10 @@ export async function connectWC(
...
   const provider = await initProvider(options, walletId);
 
-  const _isChainsState = await isChainsStale(provider, [
-    provider.chainId,
-    ...(options?.walletConnect?.optionalChains || []).map((c) => c.id),
-  ]);
+  // const _isChainsState = await isChainsStale(provider, [
+  //   provider.chainId,
+  //   ...(options?.walletConnect?.optionalChains || []).map((c) => c.id),
+  // ]);
```
Comment (jnsdls) [inline packages/thirdweb/src/wallets/wallet-connect/index.ts:76]: just remove this

File: packages/thirdweb/src/wallets/wallet-connect/index.ts [hunk 5]
```diff
@@ -52,7 +52,7 @@ const defaultWCProjectId = "08c4b07e3ad25f1a27c14a4e8cecb6f0";
 const NAMESPACE = "eip155";
 const ADD_ETH_CHAIN_METHOD = "wallet_addEthereumChain";
 
-const isNewChainsStale = true;
+// const isNewChainsStale = true;
```
Comment (jnsdls) [inline packages/thirdweb/src/wallets/wallet-connect/index.ts:55]: also just remove

File: packages/thirdweb/src/wallets/wallet-connect/index.ts [hunk 6]
```diff
@@ -238,9 +236,9 @@ async function initProvider(
 
   // disconnect the provider if chains are stale when (if not auto connecting)
   if (!isAutoConnect) {
-    const isStale = await isChainsStale(provider, chainsToRequest);
+    // const isStale = await isChainsStale(provider, chainsToRequest);
```
Comment (jnsdls) [inline packages/thirdweb/src/wallets/wallet-connect/index.ts:239]: remove

File: packages/thirdweb/src/wallets/wallet-connect/index.ts [hunk 7]
```diff
@@ -455,33 +448,33 @@ async function switchChainWC(provider: WCProvider, chain: Chain) {
...
+//   ) {
+//     return false;
+//   }
 
-  // if chain was requested earlier, then they are not stale
-  return !chains.every((id) => requestedChains.includes(id));
-}
+//   // if chain was requested earlier, then they are not stale
+//   return !chains.every((id) => requestedChains.includes(id));
+// }
```
Comment (jnsdls) [inline packages/thirdweb/src/wallets/wallet-connect/index.ts:477]: remove unused

**Review by jnsdls — APPROVED** [review]
small nitpicks and cleanup comments but otherwise LGTM

**Pull request merged**